firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -88,42 +103,37 @@ internal constructor(
     }
   }
 
-  private fun initiateSessionStart() {
-    val sessionDetails = sessionGenerator.generateNewSession()
-
-    sessionStartScope.launch {
-      val subscribers = FirebaseSessionsDependencies.getRegisteredSubscribers()
+  private suspend fun initiateSessionStart(sessionDetails: SessionDetails) {
+    val subscribers = FirebaseSessionsDependencies.getRegisteredSubscribers()
 
-      if (subscribers.isEmpty()) {
-        Log.d(
-          TAG,
-          "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
-        )
-        return@launch
-      }
-
-      if (subscribers.values.none { it.isDataCollectionEnabled }) {
-        Log.d(TAG, "Data Collection is disabled for all subscribers. Skipping this Session Event")
-        return@launch
-      }
+    if (subscribers.isEmpty()) {
+      Log.d(
+        TAG,
+        "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
+      )
+      return
+    }
 
-      Log.d(TAG, "Data Collection is enabled for at least one Subscriber")
+    subscribers.values.forEach { subscriber ->
+      // Notify subscribers, regardless of sampling and data collection state.
+      subscriber.onSessionChanged(SessionSubscriber.SessionDetails(sessionDetails.sessionId))
+    }
 
-      subscribers.values.forEach { subscriber ->
-        if (subscriber.isDataCollectionEnabled) {
-          subscriber.onSessionChanged(SessionSubscriber.SessionDetails(sessionDetails.sessionId))
-        }
-      }
+    if (subscribers.values.none { it.isDataCollectionEnabled }) {
+      Log.d(TAG, "Data Collection is disabled for all subscribers. Skipping this Session Event")
+      return
+    }
 
-      if (!sessionGenerator.collectEvents) {
-        Log.d(TAG, "Sessions SDK has sampled this session")
-        return@launch
-      }
+    Log.d(TAG, "Data Collection is enabled for at least one Subscriber")
 
-      sessionCoordinator.attemptLoggingSessionEvent(
-        SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, timeProvider)
-      )
+    if (!sessionGenerator.collectEvents) {
```

Comment:
Just validating the functionality here: Even if the session data collection is sampled, we will be generating sessions, notifying the dependencies of the sessionInfo, but just that the session will not be dispatched?

Reply:
That's right.